src/python_testing/TC_CLDIM_5_1.py

```diff
@@ -0,0 +1,242 @@
+from mobly import asserts
+
+
+class TC_CLDIM_5_1(MatterBaseTest):
+    async def read_cldim_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.ClosureDimension
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_CLDIM_5_1(self) -> str:
+        return "[TC-CLDIM-5.1] {C_STEP} and {C_SET_TARGET} Commands while Latched with DUT as Server"
+
+    def steps_TC_CLDIM_5_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep("2a", "Read FeatureMap attribute"),
+            TestStep("2b", "If MotionLatching feature is not supported, skip remaining steps"),
+            TestStep("2c", "Read AttributeList attribute"),
+            TestStep("2d", "Read LimitRange attribute"),
+            TestStep("3a", "If manual latching is required, skip steps 3b and 3c"),
+            TestStep("3b", "Send SetTarget command with Latch=True"),
+            TestStep("3c", "Manually latch the device"),
+            TestStep("4a", "If manual latching is not required, skip steps 4b to 4d"),
+            TestStep("4b", "Send SetTarget command with Latch=True"),
+            TestStep("4c", "Verify Target attribute is updated"),
+            TestStep("4d", "Wait for PIXIT.CLDIM.LatchingDuration seconds"),
+            TestStep(5, "Verify CurrentState attribute is updated"),
+            TestStep("6a", "Send C_STEP command while latched"),
+            TestStep("6b", "Send C_SET_TARGET command while latched"),
+            TestStep("7a", "Send SetTarget command with Latch=False"),
+            TestStep("7b", "Wait for PIXIT.CLDIM.LatchingDuration seconds"),
+            TestStep("7c", "Verify CurrentState attribute is updated"),
+        ]
+        return steps
+
+    def pics_TC_CLDIM_5_1(self) -> list[str]:
+        pics = [
+            "CLDIM.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_CLDIM_5_1(self):
+        asserts.assert_true('PIXIT.CLDIM.LatchingDuration' in self.matter_test_config.global_test_params,
+                            "PIXIT.CLDIM.LatchingDuration must be included on the command line in "
+                            "the --int-arg flag as PIXIT.CLDIM.LatchingDuration:<duration>")
+
+        latching_duration = self.matter_test_config.global_test_params['PIXIT.CLDIM.LatchingDuration']
+
+        if latching_duration <= 0:
+            asserts.fail("PIXIT.CLDIM.LatchingDuration must be greater than 0")
+
+        endpoint = self.get_endpoint(default=1)
+
+        # STEP 1: Commission DUT to TH (can be skipped if done in a preceding test)
+        self.step(1)
+        attributes = Clusters.ClosureDimension.Attributes
+
+        # Default values
+        min_position = 0
+        max_position = 10000  
+
```

Comment:
Why are there default values here? There are no such implied defaults in spec. Please do not use default positions